addons/account/models/account_payment.py

```diff
@@ -737,33 +742,34 @@ def _onchange_journal(self):
         invoices = self.env['account.invoice'].browse(active_ids)
         if self.journal_id and invoices:
             if invoices[0].type in ('out_invoice', 'in_refund'):
-                domain = [('payment_type', '=', 'inbound'), ('id', 'in', self.journal_id.inbound_payment_method_ids.ids)]
+                domain_payment = [('payment_type', '=', 'inbound'), ('id', 'in', self.journal_id.inbound_payment_method_ids.ids)]
             else:
-                domain = [('payment_type', '=', 'outbound'), ('id', 'in', self.journal_id.outbound_payment_method_ids.ids)]
-
-            return {'domain': {'payment_method_id': domain}}
+                domain_payment = [('payment_type', '=', 'outbound'), ('id', 'in', self.journal_id.outbound_payment_method_ids.ids)]
+            domain_journal = [('type', 'in', ('bank', 'cash')), ('company_id', '=', invoices[0].company_id.id)]
+            return {'domain': {'payment_method_id': domain_payment, 'journal_id': domain_journal}}
         return {}
 
-    @api.multi
-    def _prepare_payment_vals(self, invoice):
+    def _prepare_payment_vals(self, invoices):
         '''Create the payment values.
 
-        :param invoice: A single invoice/bill to pay.
+        :param invoices: The invoices/bills to pay. In case of multiple
+            documents, they need to be grouped by partner, bank, journal and
+            currency.
         :return: The payment values as a dictionary.
         '''
-        amount = self.env['account.payment']._compute_payment_amount(invoices=invoice, currency=invoice.currency_id)
+        amount = self.env['account.payment']._compute_payment_amount(invoices=invoices, currency=invoices[0].currency_id)
```

Comment:
WAIT: NOOOOOO

The currency ca not be one from the first invoice, don't do that my friend, currency is FROM THE PAYMENT not from the very first invoice found!,

Reply:
too late.

Reply:
@nhomar I don't agree. The invoices are grouped by currency. Then, the payment is created using the same currency of the first invoice as you can't select a custom one in the register payment wizard.